File: app/src/main/java/tech/pauly/findapet/data/models/FetchAnimalsRequest.java
package tech.pauly.findapet.data.models;

import java.util.Objects;

public class FetchAnimalsRequest {
    private final String location;
    private final Filter filter;
    private final AnimalType animalType;
    private final int lastOffset;

    public FetchAnimalsRequest(AnimalType animalType, int lastOffset, String location, Filter filter) {
        this.animalType = animalType;
        this.lastOffset = lastOffset;
        this.location = location;
        this.filter = filter;
    }

    public String getLocation() {
        return location;
    }

    public Filter getFilter() {
        return filter;
    }

    public AnimalType getAnimalType() {
        return animalType;
    }

    public int getLastOffset() {
        return lastOffset;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        FetchAnimalsRequest request = (FetchAnimalsRequest) o;
        return lastOffset == request.lastOffset && Objects.equals(location, request.location) && Objects.equals(filter, request.filter) && animalType == request.animalType;
    }

    @Override
    public int hashCode() {

        return Objects.hash(location, filter, animalType, lastOffset);
    }
}

File: app-old/src/main/java/tech/pauly/old/data/models/FetchAnimalsRequest.kt
package tech.pauly.old.data.models

open class FetchAnimalsRequest(open val animalType: AnimalType,
                               open val lastOffset: Int,
                               open val location: String,
                               open val filter: Filter) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FetchAnimalsRequest

        if (animalType != other.animalType) return false
        if (lastOffset != other.lastOffset) return false
        if (location != other.location) return false
        if (filter != other.filter) return false

        return true
    }

    override fun hashCode(): Int {
        var result = animalType.hashCode()
        result = 31 * result + lastOffset
        result = 31 * result + location.hashCode()
        result = 31 * result + filter.hashCode()
        return result
    }
}
